src/virtual_secondary/managedsecondary.cc

```diff
@@ -153,7 +158,7 @@ data::ResultCode::Numeric ManagedSecondary::install(const Uptane::Target &target
     return data::ResultCode::Numeric::kInstallFailed;
   }
 
-  image_reader(target_name)->writeToFile(sconfig.firmware_path);
+  image_reader(target)->writeToFile(sconfig.firmware_path);
   Utils::writeFile(sconfig.target_name_path, expected_target_name);
   sync();
```

Comment:
I think I've forgotten to remove this `sync` as we agreed in the scope of my last PR discussion (or it was restored somehow).

Reply:
Probably my bad restoring during multiple rounds of rebasing. It's gone now.